sn_networking/src/event/swarm.rs

```diff
@@ -80,6 +80,9 @@ impl SwarmDriver {
                 }
                 event_string = "upnp_event";
                 info!(?upnp_event, "UPnP event");
+                if let libp2p::upnp::Event::GatewayNotFound = upnp_event {
+                    panic!("UPnP is not enabled/supported on your gateway. Please rerun without the `--upnp` flag");
```

Comment:
This should be an error that we bubble up. In that case we could also later do graceful shutdowns of our swarm. `sn_networking` is a library crate, so we don't want consumers to get panics.

Reply:
We can bubble the error up, but it would be by one level and still within the `sn_networking` lib:

https://github.com/maidsafe/safe_network/blob/main/sn_networking/src/driver.rs#L735-L741

Handling this error gracefully would probably take some refactoring. Not sure if that is worth it. Thoughts?